### DISCUSSION TITLE
The audio quality of ADC acquisition speakers is particularly poor

### DISCUSSION THREAD

**Discussion by 90757665** [pschatzmann/arduino-audio-tools#875] [Q&A] [answered]
I am trying to use the built-in ADC of ESP-32 to collect audio from the walkie talkie (collect audio output from the SPK of the walkie talkie). The sampled audio can hear the sound of speech, but there is serious noise. May I ask what is causing this situation and how can I solve it

**Comment by 90757665** [comment]
![微信截图_20230529200050](https://github.com/pschatzmann/arduino-audio-tools/assets/41498674/d25c70b1-04ab-4c1c-b59f-f8633bfb164c)

**Comment by LinusHeu** [comment]
Can it be that the ADC of the ESP32 is just noisy? Googling "ESP32 ADC noise" gives a lot of results, maybe something is useful for you. Or you could look into using a separate ADC. 
Also, the source could be noisy.

**Reply by 90757665** [reply]
Yes, I also think it's due to the inaccurate ADC. I'm considering using external ADC, but I haven't found a suitable module yet

**Comment by pschatzmann** [comment, marked as answer]
Of cause an analog microphone with the internal DAC produces much more noise than an I2S microphone. But it should still be good enough to record and recognize some speech.

Your picture does not mean anything to me if you do not indicate what you were recording. I think I left enough information how to to do some tests by recording a sine wave:

https://github.com/pschatzmann/arduino-audio-tools/wiki/It's-not-working
 https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-stream/streams-adc-serial

**Reply by 90757665** [reply]
After your reminder, I generated a 440Hz, mono, 8000 sample rate wav sine wave for analog output using ESP8266. Then use esp32's adc for collection. I need your help pointing out the problem, I have been troubled for many days. I have tried changing the sampling rate and signal line, but it has not improved

440HZ sine wave diagram
![image](https://github.com/pschatzmann/arduino-audio-tools/assets/41498674/68da6c86-d0b7-4c32-9306-8a52c97b7bd0)

Esp32adc acquisition serial port printing diagram
![QQ截图20230531192057](https://github.com/pschatzmann/arduino-audio-tools/assets/41498674/4bb97b26-63a0-4127-b011-01fb7da1f878)

Audio graph viewed by computer after ADC acquisition
![image](https://github.com/pschatzmann/arduino-audio-tools/assets/41498674/b5bc232a-cd98-4759-9b03-41566f1709c7)

**Comment by 90757665** [comment]
I am using an example: base-adc-serial for adc collection. Correction data to 0

**Comment by pschatzmann** [comment]
Hmm, I would rather think this is a hardware problem. Are you sure that you use only short wires and that you don't have a custom board which has some capacitors which might mess up the signal ?

**Reply by 90757665** [reply]
I'm sure. I initially found that the signal was chaotic when I collected the walkie talkie, so I replaced it with ESP8266 for analog audio output. Now I only use the simplest environment, ESP8266 analog audio output, and use the 32 pin connection of ESP32 to connect the rx pin of ESP8266, using only a very short wire of about 5cm. I also strongly suspect that it is an issue with ADC, but as this is my first attempt to do so, I need your experience and guidance. Do you have a better recommendation for external adc modules? I can't find a suitable one

**Comment by 90757665** [comment]
I don't know how to try it now

**Comment by pschatzmann** [comment]
Maybe a module based on the ES8388 might be a better fit. There you can control the gain by software.
Apart from that you can find an ADC in the Wiki: https://github.com/pschatzmann/arduino-audio-tools/wiki/External-ADC

**Reply by 90757665** [reply]
Thank you for your help
